### DISCUSSION TITLE
How to set different margins for landscape and portrait layouts?

### DISCUSSION THREAD

**Discussion by bodzebod** [pymupdf/pymupdf4llm#70] [Q&A] [answered]
Hi, I have some PDF files which mix portrait and landscape layouts. In order to exclude some headers/footers that are not excluded automatically, I want to set margins. However, margins are different depending on the page layout.
How can I set those different margins depending on the page layout? Thank you!

**Comment by JorjMcKie** [comment, marked as answer]
You need to invoke method `to_markdown` page-wise, i.e. use the parameter `pages=[page.number]`. Then you can give each page some special treatment and / or adjust invocation parameters.
The one thing to take care of is to execute Section Header Identifications only once and provide that result to each invocation.
Here is a snippet.
```python
import pymupdf, pymupdf4llm

doc = pymupdf.open("input.pdf")
hdr_info = pymupdf4llm.IdentifyHeaders(doc)  # perform a one-time scan of font sizes
mdtext = ""
for page in doc:
    rect = page.rect  # the page rectangle
    if rect.width > rect.height:  # do some computations for margins
        margins = (...)  # landscape values
    else:
        margins = (...)  # portrait values
    mdtext += pymupdf4llm.to_markdown(doc, pages=[page.number], margins=margins, hdr_info=hdr_info)
```

**Reply by bodzebod** [reply]
Fantastic! Thank you very much!

**Reply by JorjMcKie** [reply]
I am glad it addresses your requirement.
